### PR TITLE
add sites providers

### PR DESCRIPTION
**Pull request by SabrinaTardio**
Task/Issue URL: https://app.asana.com/0/1204186595873227/1207569384066479/f

**Description**: Implements the providers for search and site suggestions showed in the contextual onboarding according to  https://app.asana.com/0/1202552961248957/1207407312592957/f

**Steps to test this PR**:
1. Check tests meet description in task https://app.asana.com/0/1202552961248957/1207407312592957/f and that they pass

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by alessandroboron** [comment]
Hey @SabrinaTardio perhaps this PR needs to be rebased or the target branch should be [sabrina/implement-dialogs](https://github.com/duckduckgo/iOS/tree/sabrina/implement-dialogs)? I can see code related to Contextual Dax Dialog but this should be only related to the Providers right?

File: DuckDuckGo/OnboardingHelpers/OnboardingSuggestedSearchesProvider.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            search = UserText.DaxOnboardingExperiment.ContextualOnboarding.tryASearchOption1English
+        } else {
+            search = UserText.DaxOnboardingExperiment.ContextualOnboarding.tryASearchOption1International
+        }
+        return ContextualOnboardingListItem.search(title: search)
+    }
+
+    private var option2: ContextualOnboardingListItem {
+        var search: String
+        if country == "us" {
```
Comment (alessandroboron) [inline DuckDuckGo/OnboardingHelpers/OnboardingSuggestedSearchesProvider.swift:66]: From doc:
_Search option 2:
if US country → mighty ducks cast
else → cast of avatar_

```
static let tryASearchOption2English = NSLocalizedString("ontextual.onboarding.try-search.option2-english", value: "cast of avatar", comment: "Search query for the cast of Mighty Ducks")
static let tryASearchOption2International = NSLocalizedString("ontextual.onboarding.try-search.option2-international", value: "mighty ducks cast", comment: "Search query for the cast of Avatar")
```

The value of the NSLocalizedString are flipped. The comments are good.

Comment (SabrinaTardio) [reply DuckDuckGo/OnboardingHelpers/OnboardingSuggestedSearchesProvider.swift:66]: Good catch!

**Review by brindy — CHANGES_REQUESTED** [review]
Seems to be a bunch of things here from other PRs and some conflicts, can you tidy it up so there's not as many lines, please?  Please ping for another review once you've done that.  Thanks!

**Comment by SabrinaTardio** [comment]
> Seems to be a bunch of things here from other PRs and some conflicts, can you tidy it up so there's not as many lines, please? Please ping for another review once you've done that. Thanks!

All changes you see now should be the actual changes. I rebased this like 3 times… -_-

There are some small changes to the view files

File: DuckDuckGo/OnboardingHelpers/OnboardingSuggestedSitesProvider.swift [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        return ContextualOnboardingListItem.site(title: site)
+    }
+
+    private var surpriseMe: ContextualOnboardingListItem {
+        let site: String
+        switch Countries(rawValue: country) {
+        case .germany: site = "https://www.duden.de/rechtschreibung/Ente"
+        case .netherlands: site = "https://www.woorden.org/woord/eend"
+        case .sweden: site = "https://www.synonymer.se/sv-syn/anka"
+        default: site = "britannica.com/animal/duck"
```
Comment (brindy) [inline DuckDuckGo/OnboardingHelpers/OnboardingSuggestedSitesProvider.swift:108]: missing protocol, not sure if that matters particularly, but the others have them

**Review by brindy — APPROVED** [review]
LGTM! one minor comment then feel free to merge

File: DuckDuckGo/OnboardingHelpers/OnboardingSuggestedSitesProvider.swift (hunk 2 above)
Comment (SabrinaTardio) [reply DuckDuckGo/OnboardingHelpers/OnboardingSuggestedSitesProvider.swift:108]: > missing protocol, not sure if that matters particularly, but the others have them

Oh yeah I did notice it now going on with more work… it is fixed in following up PR!

**Pull request merged**